guava/src/com/google/common/primitives/Bytes.java

```diff
@@ -139,6 +139,35 @@ public static int lastIndexOf(byte[] array, byte target) {
     return lastIndexOf(array, target, 0, array.length);
   }
 
+  /**
+   * Returns the start position of the last occurrence of the specified {@code target} within
+   * {@code array}, or {@code -1} if there is no such occurrence.
+   *
+   * <p>More formally, returns the greatest index {@code i} such that {@code Arrays.copyOfRange(array,
+   * i, i + target.length)} contains exactly the same elements as {@code target}.
+   *
+   * @param array the array to search for the sequence {@code target}
+   * @param target the array to search for as a sub-sequence of {@code array}
+   */
+  public static int lastIndexOf(byte[] array, byte[] target) {
+    checkNotNull(array, "array");
+    checkNotNull(target, "target");
+    if (target.length == 0) {
+      return 0;
```

Comment:
IMHO the last occurrence of an empty string starts at `array.length`.

Reply:
Yep, you're right, in this case `target.length` should be a return value

Reply:
I think you meant array.length, didn't you?

Reply:
@szarnekow shouldn't have published the comment in hurry, yes, array.length